src/MongoDB/Manager.c

```diff
@@ -267,8 +268,12 @@ PHP_METHOD(Manager, selectServer)
 	}
 
 	readPreference = phongo_read_preference_from_zval(zreadPreference TSRMLS_CC);
-	server_id = mongoc_cluster_preselect(&intern->client->cluster, MONGOC_OPCODE_QUERY, readPreference, NULL);
-	phongo_server_init(return_value, intern->client, server_id TSRMLS_CC);
+	selected_server = mongoc_topology_select(intern->client->topology, MONGOC_SS_READ, readPreference, MONGOC_SS_DEFAULT_LOCAL_THRESHOLD_MS, &error);
```

Comment:
preselect has been removed as it became redundant after internal refactoring.
Now we return a mongoc_server_description_t (on success), which contains the server_id